src/main/java/com/squareup/javapoet/Util.java

```diff
@@ -82,4 +154,23 @@ public static String join(String separator, List<String> parts) {
     result.addAll(b);
     return result;
   }
+
+  public static Set<Modifier> getMissingImplicitModifiers(Set<Modifier> modifiers,
```

Comment:
yeah, that's fine. The `requireSubsetOf()` is intended to break if an unsupported modifier is used. The `requireExactlyOne` call makes sure a necessary modifier is present.